ros1_bridge/__init__.py

```diff
@@ -245,7 +245,10 @@ def get_ros1_messages(rospack=None):
     pkgs = sorted(x for x in rosmsg.iterate_packages(rospack, rosmsg.MODE_MSG))
     for package_name, path in pkgs:
         for message_name in rosmsg._list_types(path, 'msg', rosmsg.MODE_MSG):
-            msgs.append(Message(package_name, message_name, path))
+            # iterate_packages might return ROS2 paths for already found ROS1 pkgs
+            msg = Message(package_name, message_name, path)
+            if msg not in msgs:
+              msgs.append(msg)
```

Comment:
Is it guaranteed that the first message found will be the ROS 1 version and not the ROS 2 version? If not then this check might not be enough.

Reply:
I don't think it's guaranteed, so you're right.

I've reimplented this, please take a look.